src/oralb_ble/parser.py

```diff
@@ -287,3 +308,55 @@ def _start_update(self, service_info: BluetoothServiceInfo) -> None:
         self.update_binary_sensor(
             str(OralBBinarySensor.BRUSHING), bool(state == 3), None, "Brushing"
         )
+        if state == 3:
+            self._brushing = True
+            self._last_brush = time.monotonic()
+        else:
+            self._brushing = False
+
+    def poll_needed(
+        self, service_info: BluetoothServiceInfo, last_poll: float | None
+    ) -> bool:
+        """
+        This is called every time we get a service_info for a device. It means the
+        device is working and online.
+        """
+        if not self._seen_advertisement:
+            # Never need to poll if we haven't gotten advertisement data
+            return False
```

Comment:
This isn't actually possible since we must have seen it if we got here given `service_info` with the advertisement is being passed as an arg to this function